Add back support for icons to Mac wxDataViewCheckIconTextRenderer

vadz opened the pull request:
This implements suggestion of Scott in [#17473](https://trac.wxwidgets.org/ticket/17473). I've fixed one of the problems mentioned there by just adding a space (which is ugly but looks decent enough), but I don't know what to do about the vertical alignment/cell height which is insufficient, see 
![mac_dvc](https://user-images.githubusercontent.com/146917/70867071-72711300-1f71-11ea-89ea-55452fab3400.png)

But even in the current state this is better than nothing, I think, so if you (@csomor) don't see anything egregiously wrong with this, I'd like to merge it.

vadz committed:
Show wxDataViewCheckIconTextRenderer in the sample

Instead of using separate check and icon+text columns in the list model
page, use the combined check+icon+text column to verify that it works
(which is not the case current in the native Mac version).

vadz committed:
Re-add support for icons to Mac wxDataViewCheckIconTextRenderer

Icons were not supported any longer since the switch to the native
implementation in 235e61c311ee1cc3c74a637984ca40b4ed773ae2.

Restore them now by using text attachments (thanks to Scott B for the
idea). The vertical alignment is still not right, but it's better than
nothing.

See #17473.

csomor commented:
checking out and looking at it, using an attributed string is a good way out IMHO

vadz commented:
Latest suggestion from Scott in the Trac discussion is to override `-[NSCell cellBaselineOffset]` to fix the remaining problem. I'm not sure if it's the optimal way to do it, but I think it should work.

csomor commented:
Yes, thanks, I saw that, I'll experiment ...

csomor commented:
@interface wxNSTextAttachmentCellWithBaseline : NSTextAttachmentCell
{
NSPoint _offset;
}
@end

@implementation wxNSTextAttachmentCellWithBaseline

- (void) setCellBaselineOffset:(NSPoint) offset
{
    _offset=offset;
}
- (NSPoint)cellBaselineOffset
{
    return _offset;
}

@end

csomor committed:
add baselineOffset override to better align icons

Cocoa by default aligns the cell on the text baseline, which is not suitable for our needs, solved by subclassing NSTextAttachmentCell

csomor reviewed (approved):
I'm fine with that PR, please test on your machines, whether the chosen offset is a valid solution, or whether we need to do calculations ...
